### ISSUE TITLE
In Notebooks, shared links don't have the same date range as the Notebook in cases where there are multiple versions

### ISSUE THREAD

**Issue by gefaila** [influxdata/ui#4257]
### To Reproduce

- Create a Notebook
- Set the date range to your first range (e.g. the full day of 29/03/2022)
- Share the Notebook as a Link using 
![image](https://user-images.githubusercontent.com/69196723/160830491-bd6be6da-9d69-499d-a755-42f714241fa6.png) 

- Save a version using the 
![image](https://user-images.githubusercontent.com/69196723/160830556-b4ab8d0d-03f9-4ebd-838c-ab2a5117d82a.png)

- Now set the range to a different range (e.g. the full day of 24/03/2022)

In this case the Notebook itself looks like this (the date range of the full day of 24/03/2022)
![image](https://user-images.githubusercontent.com/69196723/160830852-c11f886b-5b51-4c98-adbd-1c7aa5b43501.png)

While the shared link looks like this (!) (a combination of the two saved date ranges (24th - 29th)
![image](https://user-images.githubusercontent.com/69196723/160831111-9a942a4f-2e36-462d-826d-ea6e75919bc8.png)

### Expected Behaviour
Obviously the share link is expected to cover the same date range as the Notebook. 

The fix should be Either:
1. (BETTER) the shared link should allow previous versions to **all** be viewed
2. (EVEN BETTER) the date range should be encoded into the https like Dashboards (excellent feature) so that the same notebook can be shared with different date range links to show someone different manifestations of the same issue, problem, feature, data etc

**Comment by gefaila** [comment]
I think I did the following order
- set the date range to cover 1 day (24 th March)
- shared the link
- saved a version
- changed the current version to cover 1 day (29 th March)
- loaded the link to check

What I got was that the link displayed some weird combination of times that 
was not the original one, nor did it have a start and stop spanning the two 
dates.

After a few minutes the link went back to displaying the time-date range saved.

But honestly it would be much much much way way better if the dates were 
encoded in the address for the notebook. Obviously you can't do this for 
the link because then people would be able to explore any date range for 
that data.
But if it's the notebook link then you are clearly dealing with someone who 
is within a trusted circle.

Cheers

Andrew

**Comment by asalem1** [comment]
Hmmm super weird that it was showing some cross between the two since it’s intended to only show the saved version’s time range if there is a saved time range. It could be that due to weirdness that was introduced yesterday (and has since been resolved) things got a little funky. I can’t seem to reproduce the behavior when saving a version with a custom time range, sharing the notebook, updating the time range, then saving and visiting the shared notebook again. Would you mind verifying that the bug still exists?

**Comment by gefaila** [comment]
Yes it still does it
![image](https://user-images.githubusercontent.com/69196723/161005820-a2b02c51-4766-4359-8588-b80c74ca35a8.png)

The image shows a small insight into the bug; the date range at the top says 24th -24th.  But the data spans the range 24th to 29th!

The Notebook meanwhile displays as
![image](https://user-images.githubusercontent.com/69196723/161006034-b138ba62-ceff-47b1-b1d9-f3a37c684f19.png)

Running the dashboard makes no difference.

So the bug is still there.

**Comment by gefaila** [comment]
I think your bug is that the shared link doesn't get notified of the new **start** time

**Comment by gefaila** [comment]
Deleting and resharing the link doesn't work.
New link shows
![image](https://user-images.githubusercontent.com/69196723/161007691-18109b3f-e5b9-4357-bbb2-0cb0bea5bd41.png)

**Comment by gefaila** [comment]
Another component of the bug is this...
Notebook displays the Table like this
![image](https://user-images.githubusercontent.com/69196723/161021059-5427cee7-c234-43be-a819-e486644c53e0.png)

Shared link displays the table like this.
![image](https://user-images.githubusercontent.com/69196723/161021122-562a4415-1624-4aa7-b464-82df162f4670.png)

The Notebook used to contain the white table type. But I changed it.
I already sent the link to the customer and so don't want to change the link.
I just want the link to update as one might expect.

**Comment by gefaila** [comment]
And while we are on the subject.  Why does the white table not resize properly (columns don't compress).  Also why is it on a white background when the theme is "dark"?  It looks really poor and un-Influx.

**Comment by gefaila** [comment]
I deleted the table and recreated it.
I unshared the link and reshared it.

Notebook displays
![image](https://user-images.githubusercontent.com/69196723/161022540-4b86d0bf-ab02-4687-835e-b7a98ef2c07a.png)

Share link displays
![image](https://user-images.githubusercontent.com/69196723/161022612-c693f83a-7e7a-4c14-84ca-8ad1543aaff3.png)

**Comment by gefaila** [comment]
Yes it's fixed.  Nice one!

**Comment by gefaila** [comment]
Sorry. I closed this one by accident.
The issue still persists

**Comment by asalem1** [comment]
This is all super weird. Thanks for sharing this info. I'll dig into it today and hopefully be able to push out a fix soon. Thanks for filling these and being so thorough with documenting weirdness in the experience

**Comment by asalem1** [comment]
@gefaila we've identified the issue and are working on a fix. Apologies for the delay. We'll close this issue out when it's been merged in. Thanks again for identifying and reporting the bug

**Comment by asalem1** [comment]
@gefaila the fix should be merged and deployed. Please feel free to reach out if there are any other issues you see with this or any other feature

**Comment by gefaila** [comment]
It's not fixed.
my Notebook https://eu-central-1-1.aws.cloud2.influxdata.com/orgs/faac9e69ed04ee08/notebooks/5ff6a15f811d2000 has a timeframe of JUST 24th
![image](https://user-images.githubusercontent.com/69196723/163221337-e077c38a-7762-4620-aa32-08ea324c2dd1.png)

Yet the shared link is pointing to a different date
![image](https://user-images.githubusercontent.com/69196723/163221438-f63bd0f3-6805-4352-b27b-bad775a6ef94.png)

The confusion is caused because I have different versions of the Notebook.
What is really needed is that each version should have a different share link.

At the moment there is one link, which means that the Notebook and the share link can get out of sync.

**Comment by gefaila** [comment]
Share link is https://eu-central-1-1.aws.cloud2.influxdata.com/share/8xDQepF9DVjGjZt0fdr3Z

**Comment by asalem1** [comment]
🤔 the share link should be pointing to the last published version's image. That is to say, that if a notebook version has been published and there have been changes applied after the fact that haven't been published, those updated changes won't be applied onto to the existing share link.

Re: having multiple share links per version, unfortunately that's not currently on the roadmap for the sharing functionality, but it sounds like a great idea for a feature submission. As a workaround, one thing that can be done would be to clone a specific notebook at that version and creating a new share link. Obviously, this requires more effort than what you're suggesting, but I'm not sure if we currently have priority for iterating on the sharing functionality beyond the current scope.

> my Notebook https://eu-central-1-1.aws.cloud2.influxdata.com/orgs/faac9e69ed04ee08/notebooks/5ff6a15f811d2000 has a timeframe of JUST 24th

With regards to this comment. Is that what's currently been published?

**Comment by gefaila** [comment]
The link represents *NEITHER* version.

But that assertion follows logically from the above if you think about it.

**Comment by asalem1** [comment]
@gefaila sorry, I'm not sure I follow. Looking at the link you shared in an incognito window, it looks like the time range that's set for the notebook is `2022-03-27 22:00 - 2022-03-29 08:00`, which is what I'm seeing for the data as well. I've attached a video here of what I'm seeing on my end when I visited the shared link:

https://user-images.githubusercontent.com/19984220/169529836-2cb9a279-1abf-4731-a76f-b578d87cb1d4.mov


Sorry I keep going around this issue, I just don't have access to the data so it's hard to know what the issue is. In your comment above, you mentioned:

> The link represents NEITHER version.
> But that assertion follows logically from the above if you think about it.

Based on the above, it's my understanding that the data that's showing up on the current shared version of the notebook is neither the last published version of the notebook AND it is not currently the most recent non-published version of the two. Is that correct?

**Comment by gefaila** [comment]
This is very frustrating.  

- The NOTEBOOK is saved with a date range of  JUST 24/03/2022
![image](https://user-images.githubusercontent.com/69196723/169545457-5e24fd8f-e0a0-4d06-9152-4d9ca62759fd.png)

- You have confirmed that the SHARED LINK is displaying 2022-03-27 22:00 - 2022-03-29 08:00
![image](https://user-images.githubusercontent.com/69196723/169545850-97d9e4d0-3b9c-4970-9528-fe5feca0c9f4.png)


Questions:
**Should** the link display the same date range as the latest notebook?  
**Does** the link display the same date range as the latest notebook? 

Please compare
my Notebook https://eu-central-1-1.aws.cloud2.influxdata.com/orgs/faac9e69ed04ee08/notebooks/5ff6a15f811d2000
(24/03/2022)
**You now have acccess.**

shared link at https://eu-central-1-1.aws.cloud2.influxdata.com/share/8xDQepF9DVjGjZt0fdr3Z 
different date range of 28-29/03/2022

**Comment by gefaila** [comment]
Please do not edit the notebook because it is currently displaying the bug.
If you fiddle with it it will revert.
but then you won't be able to see the bug and you'll be even more confused.

**Comment by asalem1** [comment]
@gefaila thanks for the invitation. It looks like the most current version of the shared notebook is the same as the most recent published version of the notebook:

https://eu-central-1-1.aws.cloud2.influxdata.com/orgs/faac9e69ed04ee08/notebooks/5ff6a15f811d2000/versions/6008e789142e0000

Please see the video below:

https://user-images.githubusercontent.com/19984220/169566312-17fa64b8-2b1c-4725-91a9-e619f3c36e08.mov

As a recap:

- The most recent version of the notebook is 24/3/2022
- The most recent **published** version of the notebook is 28-03-2022 - 29-03-2022
- The most recent **published** version is the version of the notebook that is available in the shared notebook.

If there's nothing I'm missing, then this all seems to be the correct behavior based on the feature. To be clear, the shared notebooks & published notebook interactions are defined as follows:

1. If a user shares a notebook that has not be published, the shared notebook will be in sync with the **live** version of the notebook will be
2. If a user **publishes** a notebook, then the shared notebook will be in sync with the **latest published version** of the notebook

**Comment by garylfowler** [comment]
I've put this on the backlog to revisit at a future date; this confusion I think is at least partly based on my decision to call the option that was originally called "publish" to instead be called "save to revision history." I did this because I felt the term "publish" would be confusing (and still believe it would be, at least  for some users- especially those not using the shared notebook feature).   The publish feature set originally included the ability to save revisions of a notebook, but also to state which one should be used for shared notebooks, via the publishing function(which I renamed).   My apologies for the confusion, will re-visit this again at a future date.
